src/postgres-data/buildUtilities.ts

```diff
@@ -10,81 +10,46 @@
  * All portions of the code written by Atypon Systems LLC are Copyright (c) 2019 Atypon Systems LLC. All Rights Reserved.
  */
 
-import {
-  BibliographyItem,
-  Correction,
-  LibraryCollection,
-  Manuscript,
-  Model,
-  ObjectTypes,
-  Project,
-  UserProfile,
-} from '@manuscripts/json-schema'
+import { Manuscript, Model, ObjectTypes } from '@manuscripts/json-schema'
 import {
   Build,
   ContainedModel,
   ContainedProps,
   isManuscriptModel,
-  ManuscriptModel,
 } from '@manuscripts/transform'
 
 import Api from '../postgres-data/Api'
-import { ContainedIDs, ContainerIDs, state } from '../store'
+import { ContainerIDs, state } from '../store'
 import { saveWithThrottle } from './savingUtilities'
 
-const buildUtilities = (
-  getData: () => Partial<state>,
-  api: Api,
-  updateState: (state: Partial<state>) => void
+export const buildUtilities = (
+  getState: () => Partial<state>,
+  updateState: (state: Partial<state>) => void,
+  api: Api
 ) => {
   const getModel = <T extends Model>(id: string) => {
-    const data = getData()
-    if (!data.modelMap) {
+    const state = getState()
+    if (!state.modelMap) {
       return
     }
-    return data.modelMap.get(id) as T | undefined
-  }
-
-  //   const bulkUpdate = async (items: Array<ContainedModel>): Promise<void> => {
-  //     for (const value of items) {
-  //       const containerIDs: ContainerIDs = {
-  //         containerID: this.containerID,
-  //       }
-  //       if (isManuscriptModel(value)) {
-  //         containerIDs.manuscriptID = this.manuscriptID
-  //       }
-  //       await this.collection.save(value, containerIDs, true)
-  //     }
-  //   }
-
-  const bulkPersistentProjectSave = (models: ManuscriptModel[]) => {
-    // combine entire project and overwrite?
-    const onlyProjectModels = models.filter((model) => !model.manuscriptID)
-    const data = getData()
-    if (data.projectID && data.manuscriptID) {
-      api.saveProject(data.projectID, onlyProjectModels)
-    }
+    return state.modelMap.get(id) as T | undefined
   }
 
-  const bulkPersistentManuscriptSave = (models: ManuscriptModel[]) => {
-    const clearedModels = models.filter((model) => {
-      return model.objectType !== ObjectTypes.Project
-    })
-
-    const data = getData()
+  const saveModels = async (models: Model[]) => {
+    const state = getState()
 
-    if (data.projectID && data.manuscriptID) {
-      return api
-        .saveProject(data.projectID, clearedModels)
-        .then(() => {
-          return true // not sure what will be returned at this point
-        })
-        .catch((e) => {
-          return false
-        })
-    } else {
-      return Promise.reject(false)
+    if (state.projectID && state.manuscriptID) {
+      try {
+        const filtered = models.filter(
+          (m) => m.objectType !== ObjectTypes.Project
+        )
+        await api.saveProject(state.projectID, filtered)
```

Comment:
do we not need to make sure to send all the models for that document, won't API delete the models that we didn't send?

Reply:
This should have the same logic as before, just using async/await, did I miss something with the conversion?

Reply:
No, you are right. That's how it used to be